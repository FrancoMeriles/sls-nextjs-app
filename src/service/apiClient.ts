import axios, { AxiosResponse, AxiosInstance } from 'axios'

class ApiClient {
  instance: AxiosInstance
  constructor(baseURL) {
    this.instance = axios.create({
      baseURL,
    })
  }

  responseBody(response: AxiosResponse) {
    return response.data
  }

  request() {
    return {
      get: (url: string) => this.instance.get(url).then(this.responseBody),
      post: (url: string, body: unknown) => this.instance.post(url, body).then(this.responseBody),
      put: (url: string, body: unknown) => this.instance.put(url, body).then(this.responseBody),
      delete: (url: string) => this.instance.delete(url).then(this.responseBody),
    }
  }
}

export default ApiClient
