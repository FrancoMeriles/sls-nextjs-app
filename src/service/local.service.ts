import ApiClient from './apiClient'
import { CategoriesType, CategorieType } from '@base/types'

class LocalApi {
  apiClient: any
  request: any
  constructor() {
    this.apiClient = new ApiClient('http://localhost:3000/api')
    this.request = this.apiClient.request()
  }

  async getCategories(): Promise<CategoriesType[]> {
    return this.request.get('/categories')
  }

  async getCategorie(categorieId: string): Promise<CategorieType> {
    return this.request.get(`/categories/${categorieId}`)
  }
}

export default LocalApi
