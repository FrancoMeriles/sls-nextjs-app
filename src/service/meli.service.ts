import ApiClient from './apiClient'
import { CategoriesType, CategorieType } from '@base/types'

class MeliService {
  apiClient: any
  request: any
  constructor() {
    this.apiClient = new ApiClient('https://api.mercadolibre.com')
    this.request = this.apiClient.request()
  }

  async getCategories(): Promise<CategoriesType[]> {
    return this.request.get('/sites/MLA/categories')
  }

  async getCategorie(categoryId: string): Promise<CategorieType> {
    return this.request.get(`/categories/${categoryId}`)
  }

  async getProductsByCategory(categoryId: string, pageNumber: number) {
    return this.request.get('/sites/MLA/search', {
      params: {
        category: categoryId,
        offset: 50 * pageNumber,
      },
    })
  }

  async getProductById(productId: string) {
    return this.request.get(`/items/${productId}`)
  }

  async getProductsByQuery(query: string, page: number): Promise<any> {
    return this.request.get(`/sites/MLA/search`, {
      params: {
        q: query,
        offset: 50 * page,
      },
    })
  }

  async getRatedProductById(productID) {
    return this.request.get(`/reviews/item/${productID}`)
  }

  async getDescriptionByProductId(productID) {
    return this.request.get(`/items/${productID}/description`)
  }

  async getShippingOptionsByProductId(productID) {
    return this.request.get(`/items/${productID}/shipping_options`, {
      params: {
        zip_code: 5000,
      },
    })
  }
}

export default MeliService
