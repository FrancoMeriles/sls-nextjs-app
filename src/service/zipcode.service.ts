import ApiClient from './apiClient'

class ZipCode {
  apiClient: any
  request: any
  constructor() {
    this.apiClient = new ApiClient('https://ipgeolocation.abstractapi.com')
    this.request = this.apiClient.request()
  }

  async getZipCode(): Promise<any> {
    return this.request.get('/v1/', {
      params: {
        api_key: process.env.NEXT_PUBLIC_PRODUCTION_API_ZIPCODE,
      },
    })
  }
}

export default ZipCode
