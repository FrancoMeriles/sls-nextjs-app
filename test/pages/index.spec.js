import { getPage } from 'next-page-tester';
import { screen } from '@testing-library/react';
import "@base/service/local.service"

jest.mock('@base/service/local.service', () => jest.fn(() => ({
  getCategories: () => [{id: 1, name: 'test'}, {id: 2, name: 'another'}],
})));
describe('Blog page', () => {
  it('page renders and matches snapshot', async () => {
    const { serverRenderToString } = await getPage({
      route: '/',
      useDocument: true,
    });
    const { html } = serverRenderToString();
    expect(html).toMatchSnapshot();
  });
  it('renders blog page', async () => {
    const { render } = await getPage({
      route: '/',
    });

    render();
    expect(screen.getByText('Welcome to')).toBeInTheDocument();
  });
});