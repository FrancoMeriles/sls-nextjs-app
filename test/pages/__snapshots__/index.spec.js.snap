// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render and match snapshot with required props 1`] = `
<div
  className="chakra-container css-f91va8"
>
  <div
    className="css-s13uex"
  >
    <div
      className="chakra-linkbox css-10n1vic"
    >
      <a
        className="chakra-linkbox__overlay css-1hnz6hu"
        href="/categorie/1/products?page=1"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <h3>
          __FIRST_NAME__
        </h3>
        <span
          style={
            Object {
              "backgroundColor": "hsl(214,100%,50%)",
            }
          }
        />
      </a>
    </div>
    <div
      className="chakra-linkbox css-10n1vic"
    >
      <a
        className="chakra-linkbox__overlay css-1hnz6hu"
        href="/categorie/2/products?page=1"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <h3>
          __SECOND_NAME__
        </h3>
        <span
          style={
            Object {
              "backgroundColor": "hsl(214,100%,50%)",
            }
          }
        />
      </a>
    </div>
  </div>
</div>
`;
